pems-server/src/test/java/org/maroubra/pemsserver/monitoring/utsapi/WebSensorTest.java

```diff
@@ -1,29 +1,56 @@
 package org.maroubra.pemsserver.monitoring.utsapi;
 
-
+import io.reactivex.Flowable;
 import org.junit.Test;
+import org.maroubra.pemsserver.monitoring.Sensor;
+import org.maroubra.pemsserver.monitoring.SensorLog;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 
 import static com.google.common.truth.Truth.assertThat;
 
 public class WebSensorTest {
 
+
     @Test
-    public void getSensorData() {
-        WebSensor webSensor1 = new  WebSensor("wasp","ES_B_11_429_3E90","BAT");
-        WebSensor webSensor2 = new WebSensor("wasp", "ES_B_05_174_7BE3", "BAT");
-        WebSensor webSensor3 = new WebSensor("wasp","ES_B_11_429_3E90","TCA");
-        List<List<String[]>> dataList = new ArrayList<>();
-        dataList.add(webSensor1.pollSensor());
-        dataList.add(webSensor1.pollSensor());
-        dataList.add(webSensor1.pollSensor());
-
-        for (List<String[]> data : dataList) {
-            if (data != null)
-                assertThat(data.get(0)).hasLength(2);
+    public void testWebSensorStart() {
+        final Logger log = LoggerFactory.getLogger(WebSensor.class);
+
+        WebSensor.Config webSensorConfig1 = new WebSensor.Config();
+        webSensorConfig1.setId("Web001");
+        webSensorConfig1.setConfig("wasp","ES_B_05_416_7C15","BAT");
+
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl("http://eif-research.feit.uts.edu.au/api/")
+                .addConverterFactory(JacksonConverterFactory.create())
+                .build();
+        UtsWebApi webApi = retrofit.create(UtsWebApi.class);
```

Comment:
As mentioned previously, we should probably mock dependencies we don't own, and just assume both positive and negative case. Take a look at [mockito](http://site.mockito.org/).